### DISCUSSION TITLE
Is there an OpenSSL API available for AES-GMAC operation

### DISCUSSION THREAD

**Discussion by Yogaraj-Alamenda** [openssl/openssl#24393] [Q&A] [answered]
Is there an OpenSSL API available for AES-GMAC operation ? or is it through AES-GCM with plaintext zero only ?

**Comment by mattcaswell** [comment, marked as answer]
> Is there an OpenSSL API available for AES-GMAC operation ?

AES in combination with GMAC is AES-GCM - but I think you mean GMAC as a standalone operation. In which case, yes. You can use the EVP_MAC APIs:

https://www.openssl.org/docs/man3.3/man7/EVP_MAC-GMAC.html
https://www.openssl.org/docs/man3.3/man3/EVP_MAC_init.html

**Reply by Yogaraj-Alamenda** [reply]
Thanks for the response! Wanted to double confirm
